### ISSUE TITLE
No data in Sessions Stats 

### ISSUE THREAD

**Issue by agenceKanvas** [Pierre-Lannoy/wp-sessions#5]
**Describe the bug**
Even with Decalog installed and stats activated, I don't have any data in the stat page.

![Capture d’écran 2024-02-19 à 11 14 16](https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/0ad0743f-cca9-40e7-85c1-e46ba647736b)

I do have sessions here : 
![Capture d’écran 2024-02-19 à 11 17 59](https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/2068d286-7030-40fc-bbe3-8e411bfab232)

And of course the options is checked : 
![Capture d’écran 2024-02-19 à 11 19 35](https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/931e6ce9-78a4-461b-a5d7-339cceaf306f)

Thanks !

**Comment by Pierre-Lannoy** [comment]
Hello  :)
Since how many time do you have activated analytics?
As you've installed DecaLog (which is not mandatory for Sessions), can you tell me what you see if you run the `wp log metrics dump` wp-cli command (only for `wordpress_plugin_sessions*` metrics)?

**Comment by agenceKanvas** [comment]
Anaylics since at least 1 or 2 weeks. 

For the dump, I've attached the file : 
[dump.txt](https://github.com/Pierre-Lannoy/wp-sessions/files/14330285/dump.txt)

**Comment by Pierre-Lannoy** [comment]
Do you have some errors in your error log or DecaLog loggers?

**Comment by agenceKanvas** [comment]
Nothing related to Sessions or that seems related to sessions.

**Comment by Pierre-Lannoy** [comment]
OK :(
So I'm going to need a bit more context. You know: versions, framework, plugins, how Sessions was installed, etc.
Thanks

**Comment by agenceKanvas** [comment]
Plugins : 
[dump.txt](https://github.com/Pierre-Lannoy/wp-sessions/files/14332078/dump.txt)
Last version of wordpress. Session was installed via the plugin installer 
If needed, I also am on the WordPressFR slack !

**Comment by agenceKanvas** [comment]
<img width="1330" alt="Capture d’écran 2024-02-22 à 11 45 13" src="https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/95d52c70-6fa0-4cc4-932d-64c899d606d2">
Strange, I DO have data for the device stats

**Comment by Pierre-Lannoy** [comment]
Hello!
I've asked several users who use this plugin to run tests, and we've been unable to reproduce it at all.
Can you verify you have the following table in your db: `xxxx_sessions_statistics`?

**Comment by agenceKanvas** [comment]
Hello 8 I do, and it's empty

**Comment by agenceKanvas** [comment]
![Capture d’écran 2024-02-23 à 15 13 15](https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/423f4213-3834-4393-b4ce-4cf72570e701)
![Capture d’écran 2024-02-23 à 15 12 58](https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/2a68ee15-ddc2-4f6d-916c-40c10276d15b)
If that helps, that's what I have in the decalog options

**Comment by Pierre-Lannoy** [comment]
Unfortunately, DecaLog has nothing to do with stats in Sessions plugin…

For now, the only reason I can think of such an issue is about `shutdown` WP hook. Something which break its execution…
If you ok with that, can you try to modify the file `wp-content/plugins/sessions/includes/features/class-schema.php` at line 56 an replace
`add_action( 'shutdown', [ self::class, 'write' ], 11, 0 );`
by 
`add_action( 'shutdown', [ self::class, 'write' ], PHP_INT_MIN, 0 );`
After that (and maybe purging opcache) do you see something in the stats table?

**Comment by agenceKanvas** [comment]
Still empty. I logged out and logged in just in case

**Comment by agenceKanvas** [comment]
I installed IP Locator and I see the first session in the table !

**Comment by Pierre-Lannoy** [comment]
Ah!
So, let's give it a little time to see if the table continues to fill up...

**Comment by Pierre-Lannoy** [comment]
Hello !
Is everything working smoothly now?

**Comment by agenceKanvas** [comment]
Stats are back. Thanks a lot. Now I have to work on them and understand how to improve everything !

**Comment by agenceKanvas** [comment]
Guess who's back :) Another website, another server. Sessions, IP and device installed. 
No stat for sessions but some for devices.
Devices :
<img width="1340" alt="Capture d’écran 2024-05-22 à 10 34 07" src="https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/2998e875-e9f2-4d31-8b76-dc7e97b86810">

Sessions (same period) : 
<img width="1347" alt="Capture d’écran 2024-05-22 à 10 34 23" src="https://github.com/Pierre-Lannoy/wp-sessions/assets/190149/d9af68ae-511e-445d-805a-87a307dab878">

**Comment by Pierre-Lannoy** [comment]
That's definitely a strange issue. Is this website on the same hosting type as the last one?

**Comment by agenceKanvas** [comment]
Not at all. The first is on a dedicated server and the other one (this one) on a cloudways/digital ocean

**Comment by Pierre-Lannoy** [comment]
And is there something similar between both of them? Common framework, (mu-)plugins, cache, cdn, etc.

**Comment by agenceKanvas** [comment]
WooCommerce, WooCommerce Subscriptions, WooCommerce Memberships, ACF Pro, Redis i think, Stripe, Gravity Forms.
These are the main plugin that are at least common on both

**Comment by Pierre-Lannoy** [comment]
Could you try ton install IP Locator to see if it's modifying something (like the other site)

**Comment by agenceKanvas** [comment]
Ah ah, already tried :)

**Comment by Pierre-Lannoy** [comment]
Have you tried to deactivate (or disable) all caching mechanisms (like redis, apcu, etc.) to see if it changes something?
Could you confirm, too, the Sessions analytics db tables are created?
